mongo/client.go

```diff
@@ -666,7 +729,12 @@ func (c *Client) configureCrypt(opts *options.AutoEncryptionOptions) error {
 		bypass = *opts.BypassAutoEncryption
 	}
 	kr := keyRetriever{coll: c.keyVaultColl}
-	cir := collInfoRetriever{client: c}
+	var cir collInfoRetriever
+	if c.metadataClient != nil {
```

Comment:
Runtime asserts aren't really a thing because the best we could do is panic and a bug that caused `metadataClient` to be accessed when it's nil would result in a panic anyway, so they don't provide much value. The new check looks good. Can you also verify that we have tests that use `bypassAutoEncryption=true`?

Reply:
Makes sense, thanks for the explanation. And yes, the new deadlock tests cases 3,4,7,8 all use `bypassAutoEncryption=true`